misc/l10n_testing.md

```diff
@@ -12,46 +12,114 @@ Other docs will cover how the Mozilla approach to the first two stages of the re
+5. Once you’ve found your translation, if you see a problem with it, you return to Pontoon to make any necessary corrections.
 
-Pontoon simplifies these steps for some Web projects by enabling the in-context editor (aka WYSIWYG editor). This allows you to see your translations in-context the moment you create them in Pontoon, cutting out the 30 minutes to 24 hour waiting period to test.
+Pontoon simplifies these steps for some Web projects by enabling the in-context editor (aka WYSIWYG editor). This allows you to see your translations in-context the moment you create them in Pontoon, cutting out the 30 minutes to 24 hours waiting period to test. Most projects do not have this enabled, however, so it is still important to know the typical testing process for your projects.
 
-You might be wondering, "what types of bugs am I supposed to be looking for when I test?" Good question :-) Here's a list of common l10n bugs with screenshots that illustrate the problems in context.
+You might be wondering, “what sort of problems am I supposed to be looking for when I test?” Good question :-) Here’s a list of common l10n bugs with screenshots that illustrate the problems in context.
 
 ## Common l10n bugs
 
 ### Encoding and fonts
 
+Character encoding errors occur when the browser can’t find the correct symbol to display for a character. Normally, you will recognize these errors when you see Unicode replacement characters (�) instead of letters or radicals in your language. These can often be resolved by either changing your browser’s default encoding or by asking developers to use a Unicode-enabled font on the site.
+
+![Encoding bug](/assets/images/l10n_errors/Pontoon_encoding.png)
+![Font bug](/assets/images/l10n_errors/font_change_bug.png)
+
 ### Truncation
 
+Truncation happens when a translation is too long to fit in its corresponding space in the user interface. This error can cause problems with transmitting the right meaning to the user by hiding part of the translation from view. It can be corrected one of two ways:
+* the developer increases the UI space to accommodate longer translations,
+* the localizer creates a shorter translation that captures the primary message but may not capture secondary or tertiary messages.
+
+![Truncation bug](/assets/images/l10n_errors/truncation.png)
+
 ### String concatenation and empty strings
 
+String concatenation errors occur when a developer takes one string, splits it into multiple strings, and tells their code to display each new string one after another. This allows developers to reuse some parts of strings in multiple places in the user interface, but often the context is different enough between the multiple places its being used that the translation is correct in one place, but bad in another. This is fixed by requesting that developers not use string concatenation when putting strings in the UI.
```

Comment:
Previous comment is still valid.

I don't agree with this paragraph.

String concatenation is used in XUL to overcome technical limitations, like having a link in the middle of the sentence. It's not related to string reuse (that's a different problem that should be mentioned). It also doesn't explain why we need empty strings.

What you're describing here is reusing strings in a different context, which is an issue, but a different one.

in case, its being -> it's